### PR TITLE
Update LPMS to ffmpeg 7

### PR DESCRIPTION
**Pull request by j0sh**
**What does this pull request do? Explain your changes. (required)**
Updates LPMS to ffmpeg 7

**Specific updates (required)**

- Update LPMS in go.mod
- Swap install_ffmpeg.sh with LPMS version

**How did you test each of these updates (required)**

Unit tests

**Does this pull request close any open issues?**

**Checklist:**

- [x] Read the [contribution guide](./CONTRIBUTING.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: install_ffmpeg.sh [hunk 1]
```diff
@@ -1,231 +1,3 @@
...
-    # If configure fails, then print the last 100 log lines for debugging and exit.
-fi
-
-if [[ ! -e "$ROOT/ffmpeg/libavcodec/libavcodec.a" || $BUILD_TAGS == *"debug-video"* ]]; then
-  cd "$ROOT/ffmpeg"
-  make -j$NPROC
-  make -j$NPROC install
-fi
+echo 'WARNING: downloading and executing lpms/install_ffmpeg.sh, use it directly in case of issues'
+curl https://raw.githubusercontent.com/livepeer/lpms/d9c78b62effdb4f5c8fc438b6033da1090d04a03/install_ffmpeg.sh | bash -s $1
```
Comment (leszko) [inline install_ffmpeg.sh:3]: This seems to be related to the specific commit, why not to use the `master`?

**Review by leszko — COMMENTED** [review]
Thanks for the PR @j0sh 

My understanding is that we want to test it first with 1 O, before merging into `master`, is that correct? If that's the case, then let's wait with merging this one.

One additional note, wdyt about merging the repos: go-livepeer and lpms? I guess that would make it simpler to handle.

File: install_ffmpeg.sh (hunk 1 above)
Comment (Tibia2000) [reply install_ffmpeg.sh:3]: FYI I ve been using this version on my Orchestrator for 5 days now and had no issues.

**Comment by j0sh** [comment]
> My understanding is that we want to test it first with 1 O, before merging into master, is that correct? If that's the case, then let's wait with merging this one.

Sure, whatever you are comfortable with. Note that this update does not change the ffmpeg output (modulo bug fixes), so the network will remain compatible with older orchestrators. It will not be necessary to do a hard switch-over to newer orchestrators.

> One additional note, wdyt about merging the repos: go-livepeer and lpms? I guess that would make it simpler to handle.

I don't mind the separation. It's been that way since day 1 and there is a lot of git / github history baked into the LPMS repo which really helps with "git archaeology" to figure out why things are the way they are. Also it makes it easier to use LPMS outside of go-livepeer and having a separate repo helps enforce that boundary. Personally, it is kind of nice to work on LPMS by itself without having to pull in go-livepeer.

File: install_ffmpeg.sh (hunk 1 above)
Comment (j0sh) [reply install_ffmpeg.sh:3]: > This seems to be related to the specific commit, why not to use the master? 

Reproducibility - the same reason tools like go mod are pinned to specific versions. Ideally we would be able to check out any go-livepeer commit and get the exact version of ffmpeg that commit depends on.

Note that we only have to update this if the LPMS install_ffmpeg.sh script changes. If LPMS is updated without any changes to `install_ffmpeg.sh`, then we don't have to actually change anything.

We could have a script or some tooling to detect mismatches (eg, must match the latest commit to `lpms/install_ffmpeg.sh` at the specific LPMS checkout which this version of go-livepeer depends on) but that can be done later if mismatches become an issue. Even better would be a more suitable build tool that is able to handle third-party dependencies, eg bazel, but that is a completely different issue.

**Comment by Tibia2000** [comment]
@stronk-dev I am really curious about netint performance. Their servers are very dense when it comes to stream capacity. I think that would enable easy data-center deployments in the future.

**Comment by leszko** [comment]
> > My understanding is that we want to test it first with 1 O, before merging into master, is that correct? If that's the case, then let's wait with merging this one.
> 
> Sure, whatever you are comfortable with. Note that this update does not change the ffmpeg output (modulo bug fixes), so the network will remain compatible with older orchestrators. It will not be necessary to do a hard switch-over to newer orchestrators.

Ok, then, please merge it when you do the testing in staging with @emranemran 

> 
> > One additional note, wdyt about merging the repos: go-livepeer and lpms? I guess that would make it simpler to handle.
> 
> I don't mind the separation. It's been that way since day 1 and there is a lot of git / github history baked into the LPMS repo which really helps with "git archaeology" to figure out why things are the way they are. Also it makes it easier to use LPMS outside of go-livepeer and having a separate repo helps enforce that boundary. Personally, it is kind of nice to work on LPMS by itself without having to pull in go-livepeer.

I'll open the discussion in Discord.

**Review by leszko — APPROVED** [review]

**Comment by j0sh** [comment]
This upgrade passes the LPMS test suite without significant changes to the output, including timestamp handling, so we should be pretty confident this is backwards-compatible with older orchestrators. 

Several orchestrators seem to have been testing and have not reported issues directly related to this upgrade.

That being said, there are a couple things that might be worth testing:

- Test with blended orchestrators, eg having the broadcaster alternate between upgraded and non-upgraded orchestrators
- Visually confirm the output, preferably with a variety of inputs (RTMP, WebRTC, VOD, clips). Preferably use something like ffplay that will log any obvious issues with the transcoded segments
- Or use a separate tool to validate the stream, eg Apple mediastreamvalidator
- There are also fixes for intra-only mode (GOP=1) and FPS pass through (FPS=0) so it would be good to double check these with real streams

From Emran in Discord:

> Off the top of my head, I think testing the following workflows using the Studio website would give us coverage:
> * Livestreaming
> * VOD
> * Recordings
> * Clipping
> * Customer mobile and desktop streaming (Shih-Yu can help with this)

**Comment by thomshutt** [comment]
@j0sh @leszko I think at this point, I'm confident enough that we should just go ahead and merge + release this. Any objections?

**Pull request merged**